### ISSUE TITLE
RouteMatrix : Deserialization fails when the status code of any matrix item is not 200

### ISSUE THREAD

**Issue by dlarkin77** [perfahlen/AzureMapsRestServices#41]
In Postman, a request to https://atlas.microsoft.com/route/matrix/sync/json?subscription-key=MY-KEY&api-version=1.0 with this body:

```yaml
{
  "origins": {
    "type": "MultiPoint",
    "coordinates": [
      [
        47.5765939,
        10.7366205
      ]
    ]
  },
  "destinations": {
    "type": "MultiPoint",
    "coordinates": [
      [
        51.422412,
        7.28118
      ],
      [
        47.9055424,
        8.1057681
      ]
    ]
  }
}
```
gets this response:
```yaml
{
    "formatVersion": "0.0.1",
    "matrix": [
        [
            {
                "statusCode": 400,
                "response": "Engine error while executing route request: MAP_MATCHING_FAILURE: Destination (7.28118, 51.4224)",
                "detailedError": {
                    "message": "Engine error while executing route request: MAP_MATCHING_FAILURE: Destination (7.28118, 51.4224)",
                    "code": "MAP_MATCHING_FAILURE"
                }
            },
            {
                "statusCode": 200,
                "response": {
                    "routeSummary": {
                        "lengthInMeters": 631690,
                        "travelTimeInSeconds": 62548,
                        "trafficDelayInSeconds": 0,
                        "trafficLengthInMeters": 0,
                        "departureTime": "2021-07-02T16:37:30+03:00",
                        "arrivalTime": "2021-07-03T09:59:58+03:00"
                    }
                }
            }
        ]
    ],
    "summary": {
        "successfulRoutes": 1,
        "totalRoutes": 2
    }
}
```
Note that there is only one successful route.

Using the latest code in master, calling GetRouteMatrix and then GetRouteMatrixResult with the same origins and destinations as above throws this exception:

```
System.AggregateException
One or more errors occurred. (The JSON value could not be converted to AzureMapsToolkit.Common.RouteMatrixRouteSummary. Path: $.matrix[0][0].response | LineNumber: 0 | BytePositionInLine: 163.)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result()
   at AzureMapsToolkit_Core_Test.UnitTest.blahResult() in C:\Users\larkind\Desktop\AzureMapsRestServices-master\AzureMapsToolkit-Core-Test\UnitTest.cs:line 1172
   at AzureMapsToolkit_Core_Test.UnitTest.blah() in C:\Users\larkind\Desktop\AzureMapsRestServices-master\AzureMapsToolkit-Core-Test\UnitTest.cs:line 1146

System.Text.Json.JsonException
The JSON value could not be converted to AzureMapsToolkit.Common.RouteMatrixRouteSummary. Path: $.matrix[0][0].response | LineNumber: 0 | BytePositionInLine: 163.
   at System.Text.Json.ThrowHelper.ThrowJsonException_DeserializeUnableToConvertValue(Type propertyType)
   at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.JsonPropertyInfo`1.ReadJsonAndSetMember(Object obj, ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.Converters.IEnumerableDefaultConverter`2.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, TCollection& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.Converters.IEnumerableDefaultConverter`2.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, TCollection& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.JsonPropertyInfo`1.ReadJsonAndSetMember(Object obj, ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadCore[TValue](JsonConverter jsonConverter, Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadCore[TValue](Utf8JsonReader& reader, Type returnType, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Deserialize[TValue](String json, Type returnType, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Deserialize[TValue](String json, JsonSerializerOptions options)
   at AzureMapsToolkit.Common.BaseServices.GetData[T](HttpClient client, String url) in C:\Users\larkind\Desktop\AzureMapsRestServices-master\AzureMapsRestToolkit\AzureMapsRestToolkit\Common\BaseServices.cs:line 238
   at AzureMapsToolkit.AzureMapsServices.GetRouteMatrixResult(String url) in C:\Users\larkind\Desktop\AzureMapsRestServices-master\AzureMapsRestToolkit\AzureMapsRestToolkit\AzureMapsServices.cs:line 726
```

This unit test reproduces the exception above:

```
[Fact]
public void blah()
{
	var am = new AzureMapsToolkit.AzureMapsServices(_KEY);

	var origins = new List<Coordinate> {
		new Coordinate { Longitude = 47.5765939, Latitude = 10.7366205 }
	};

	var destinations = new List<Coordinate> {
		new Coordinate { Longitude = 51.422412, Latitude = 7.28118 },
		new Coordinate { Longitude =  47.9055424, Latitude = 8.1057681}
	};

	var (ResultUrl, ex) = am.GetRouteMatrix(new RouteMatrixRequest
	{
		RouteType = RouteType.Fastest
	}, origins, destinations).Result;

	Assert.Null(ex);

	Assert.NotNull(ResultUrl);

	blahResult();
}

public void blahResult()
{
	var am = new AzureMapsToolkit.AzureMapsServices(_KEY);

	var req = new RouteMatrixRequest
	{
		RouteType = RouteType.Fastest
	};

	var origins = new List<Coordinate> {
		new Coordinate { Longitude = 47.5765939, Latitude = 10.7366205 }
	};

	var destinations = new List<Coordinate> {
		new Coordinate { Longitude = 51.422412, Latitude = 7.28118 },
		new Coordinate { Longitude =  47.9055424, Latitude = 8.1057681}
	};

	var (ResultUrl, _) = am.GetRouteMatrix(req, origins, destinations).Result;

	var matrixResponse = am.GetRouteMatrixResult(ResultUrl).Result;

	Assert.Null(matrixResponse.Error);
	Assert.NotNull(matrixResponse.Result);
}
```

I **think** understand why this is happening, I'm just not sure how to fix it :)

**Comment by perfahlen** [comment]
@dlarkin77 fixed in latest release (7.0.0). This example works

```
static void Main(string[] args)
        {

            var azureMaps = new AzureMapsServices("<AM KEY>");

            var routeMatrixRequest = new RouteMatrixRequest
            {
                RouteType = RouteType.Fastest,
                TravelMode = TravelMode.Car
            };

            var origins = new List<Coordinate> {
                new Coordinate { Longitude = 10.7366205, Latitude =  47.5765939}
            };


            var destinations = new List<Coordinate> {
                new Coordinate { Latitude = 51.422412, Longitude = 7.28118 },
                new Coordinate { Latitude =  47.9055424, Longitude = 8.1057681}
            };

            var (matrixResultUrl, ex) = azureMaps.GetRouteMatrix(
                routeMatrixRequest,
                origins,
                destinations
            ).Result;

            var response = azureMaps.GetRouteMatrixResult(matrixResultUrl).Result;

        }
```

**Comment by perfahlen** [comment]
For some time ago, this library moved to .Net v 5.
